test/Microsoft.Sbom.Api.Tests/Executors/LicenseInformationFetcherTests.cs

```diff
@@ -139,6 +139,24 @@ public void ConvertComponentToListForApi_Crate()
         Assert.AreEqual("crate/cratesio/-/cratepackage/1.0.0", listOfComponentsForApi[0]);
     }
 
+    [TestMethod]
+    public void ConvertComponentToListForApi_Go()
+    {
+        var licenseInformationFetcher = new LicenseInformationFetcher(mockLogger.Object, mockRecorder.Object, mockLicenseInformationService.Object);
+
+        var scannedComponents = new List<ScannedComponent>
+        {
+            new ScannedComponent
+            {
+               Component = new GoComponent("example.com/namespace/pkgname", "1.0.0")
+            }
+        };
+
+        var listOfComponentsForApi = licenseInformationFetcher.ConvertComponentsToListForApi(scannedComponents);
+
+        Assert.AreEqual("go/golang/example.com%2fnamespace/pkgname/1.0.0", listOfComponentsForApi[0]);
```

Comment:
This is throwing an Exception. Running your unit test locally, the `case` statement on line 74 of `LicenseInformationFetcher` didn't get hit because `componentType` is parsed as "golang" instead of "go". As a result, the list of components is empty and the unit test throws an Exception when trying to access the data it contains.

You might consider running your modified tool locally against a go project, setting a breakpoint at `LicenseInformationFetcher.ConvertComponenetsToListForApi`, and working out the details with your actual project, then trying to recreate that live sample in the unit test. Once you have that, you should be able to replace "real" data (like "stretchrcom") with fake data (like "examplecom") and end up with a unit test that appropriate pins the behavior of the newly added code.

Reply:
I was able to test it by running the docker image locally against a respository. 
![image](https://github.com/user-attachments/assets/681ce29c-10ad-4795-ad0b-8db23e04c8e7)
Which looks to be correct based on the [clearly defined page](https://clearlydefined.io/definitions/go/golang/filippo.io/edwards25519/v1.1.0).

There is still an issue where nested packages such as `[github.com/jackc/pgx/v5](pkg:golang/github.com/jackc/pgx/v5@v5.7.1)` are not returning licence information. I am not sure how to handle this case yet as there could be any number of path segments that are not part of the package.

Using the example above:
- package name is: `github.com/jackc/pgx`
- version: `v5.7.1`
- sub package/folder: `v5`.

Using this pattern is very common in Golang so I will try and come up with a solution. If you have any suggestions let me know.

Reply:
Sadly, running against a local repo doesn't scale well to unit tests. We could have a test case that calls out to an external source, but doing that in a pipeline has become a security concern in the last couple of months. We could also consider having a small package in the repo, but that seems a bit sketchy. There may be a world where we can make the external call when running locally, but short-circuit it in the pipeline. This could potentially be done by checking for the presence of an environment variable (maybe `BUILD_BUILDID`) that is always set in the pipeline but that should not typically be set in a local environment. If it's set, then we're in a build pipeline and the test should be skipped.

Reply:
Strictly speaking, hitting the external site is an integration test. For the unit test, you could just use the inputs that would be used in your previous example of https://clearlydefined.io/definitions/go/golang/github.com%2fstretchrcom/testify/v1.9.1-0.20241006154923-2fc4e3939413. This code is just string conversion, so this should be sufficient. No need to worry about calling to an actual external resource

Reply:
The testing I did locally was only to verify that the tool worked when used, I wasn't suggesting testing against a local repo.

Sorry for the delay getting back to you.